recipes/sdl/all/conanfile.py

```diff
@@ -453,5 +459,5 @@ def package_info(self):
         # Workaround to avoid unwanted sdl::sdl target in CMakeDeps generator
         self.cpp_info.set_property(
             "cmake_target_name",
-            "SDL2::{}".format("SDL2main" if self.options.sdl2main else sdl2_cmake_target),
+            "SDL2::{}".format("SDL2main" if self.options.sdl2main else "SDL2"),
```

Comment:
> If sdl2main == False will Conan generate two different targets SDL2::SDL2?

No, the "global" one will be skipped, because this target is always the last one created by CMakeDeps (or cmake_find_package*) and only if it does not already exist.